lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -639,12 +643,15 @@ def _build_student_data(cls, user_id, course_key, usage_key_str):
                     response['location'] = ' > '.join(path)
                     # A machine-friendly location for the current block
                     response['block_key'] = str(block_key)
+                    user_data = generated_report_data.get(response['username'], {})
+                    response.update(user_data)
+                    student_data_keys = student_data_keys.union(user_data.keys())
                 if max_count is not None:
                     max_count -= len(responses)
                     if max_count <= 0:
                         break
 
-        return student_data
+        return student_data, list(student_data_keys)
```

Comment:
The docstring says this returns a `Set` but here it's returning a `list`

Reply:
Updated the docstring.